src-tauri/src/periodic/config.rs

```diff
@@ -0,0 +1,185 @@
+use std::{collections::HashSet, str::FromStr, time::Duration};
+use tauri::{AppHandle, Manager, State, Url};
+use tokio::time::sleep;
+
+use crate::{
+    appstate::AppState,
+    commands::{device_config_to_location, update_instance},
+    database::{
+        models::{Id, NoId},
+        DbPool, Instance, Location, WireguardKeys,
+    },
+    error::Error,
+    events::{CONFIG_CHANGED, INSTANCE_UPDATE},
+    proto::{DeviceConfigResponse, InstanceInfoRequest, InstanceInfoResponse},
+};
+
+const INTERVAL_SECONDS: Duration = Duration::from_secs(30);
+const POLLING_ENDPOINT: &str = "/api/v1/poll";
+
+/// Periodically retrieves and updates configuration for all [`Instance`]s.
+/// Updates are only performed if no connections are established to the [`Instance`],
+/// otherwise event is emmited and UI message is displayed.
+pub async fn poll_config(handle: AppHandle) {
+    let state: State<AppState> = handle.state();
+    let pool = state.get_pool();
+    loop {
+        let Ok(instances) = Instance::all(&pool).await else {
+            error!("Failed to retireve instances, retrying in {}s", INTERVAL_SECONDS.as_secs());
+            sleep(INTERVAL_SECONDS).await;
+            continue;
+        };
+        debug!(
+            "Polling configuration updates for {} instances",
+            instances.len(),
+        );
+        for instance in &instances {
+            if let Err(err) = poll_instance(&pool, instance, handle.clone()).await {
+                error!(
+                    "Failed to retrieve instance {}({}) config: {}",
+                    instance.name, instance.id, err
+                );
+            } else {
+                debug!(
+                    "Retrieved config for instance {}({})",
+                    instance.name, instance.id,
+                );
+            }
+        }
+        info!(
+            "Retrieved configuration for {} instances, sleeping {}s",
+            instances.len(),
+            INTERVAL_SECONDS.as_secs(),
+        );
+        sleep(INTERVAL_SECONDS).await;
+    }
+}
+
+/// Retrieves configuration for given [`Instance`].
+/// Updates the instance if there are no active connections, otherwise displays UI message.
+pub async fn poll_instance(
+    pool: &DbPool,
+    instance: &Instance<Id>,
+    handle: AppHandle,
+) -> Result<(), Error> {
+    // Query proxy api
+    let request = build_request(pool, instance).await?;
+    let url = Url::from_str(&instance.proxy_url)
+        .and_then(|url| url.join(POLLING_ENDPOINT))
+        .map_err(|_| {
+            Error::InternalError(format!(
+                "Can't build polling url: {}/{}",
+                instance.proxy_url, POLLING_ENDPOINT
+            ))
+        })?;
+    let response = reqwest::Client::new().post(url).json(&request).send().await;
```

Comment:
```suggestion
    let response = reqwest::Client::new().post(url).json(&request).timeout(Duration::from_secs(5)).send().await;
```
maybe? Not sure if there is any default timeout:
https://docs.rs/reqwest/latest/reqwest/struct.RequestBuilder.html#method.timeout
https://docs.rs/reqwest/latest/reqwest/struct.ClientBuilder.html#method.timeout